homeassistant/components/diagnostics/__init__.py

```diff
@@ -62,33 +83,35 @@ async def async_get_device_diagnostics(
 
 async def _register_diagnostics_platform(
     hass: HomeAssistant, integration_domain: str, platform: DiagnosticsProtocol
-):
+) -> None:
     """Register a diagnostics platform."""
-    hass.data[DOMAIN][integration_domain] = {
-        DiagnosticsType.CONFIG_ENTRY.value: getattr(
-            platform, "async_get_config_entry_diagnostics", None
-        ),
-        DiagnosticsSubType.DEVICE.value: getattr(
-            platform, "async_get_device_diagnostics", None
-        ),
-    }
+    diagnostics_data: DiagnosticsData = hass.data[DOMAIN]
+    diagnostics_data.platforms[integration_domain] = DiagnosticsPlatformData(
+        getattr(platform, "async_get_config_entry_diagnostics", None),
+        getattr(platform, "async_get_device_diagnostics", None),
+    )
 
 
 @websocket_api.require_admin
 @websocket_api.websocket_command({vol.Required("type"): "diagnostics/list"})
 @callback
 def handle_info(
     hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
-):
+) -> None:
     """List all possible diagnostic handlers."""
+    diagnostics_data: DiagnosticsData = hass.data[DOMAIN]
     connection.send_result(
         msg["id"],
         [
             {
                 "domain": domain,
-                "handlers": {key: val is not None for key, val in info.items()},
+                "handlers": {
+                    DiagnosticsType.CONFIG_ENTRY: info.config_entry_diagnostics
+                    is not None,
```

Comment:
The line break makes it hard to read. 🙈

Reply:
OK, I shuffled the code around a bit to avoid the line break.